lib/requester/requester.js

```diff
@@ -228,48 +228,25 @@ _.assign(Requester.prototype, /** @lends Requester.prototype */ {
              * @param {Object[]} headers
              */
             addMissingRequestHeaders = function (headers) {
-                var header,
-                    lowerCasedKey;
-
-                _.forOwn(headers, function (value, key) {
-                    // @todo handle disabled headers
-                    header = request.headers.one(key);
-                    lowerCasedKey = _.lowerCase(key);
-
-                    // neither request nor runtime overwrites user-defined headers
-                    // so instead of upsert, just add missing headers to the list
-                    // with system: true flag.
-                    if (!header || header.disabled) {
-                        request.headers.add({
-                            key: key,
-                            value: value,
-                            system: true
-                        });
-
-                        return;
-                    }
-
-                    // bail out if
-                    // 1. header is not one of overwritten headers
-                    // 2. overwritten header value is not updated
-                    if (!(OVERWRITTEN_HEADERS[lowerCasedKey] && value !== header.value)) {
-                        return;
-                    }
+                _.forEach(headers, function (header) {
+                    // update headers which gets overwritten by the requester
+                    if (OVERWRITTEN_HEADERS[header.key.toLowerCase()]) {
+                        let lowerCasedKey = header.key.toLowerCase();
+
+                        // in case of duplicate headers, remove all the duplicates
+                        // just keep the last item in the list.
+                        if (Array.isArray(_.get(request.headers, ['reference', lowerCasedKey]))) {
+                            request.headers.remove(function (h) {
+                                return _.lowerCase(h.key) === lowerCasedKey && h !== header;
+                            });
+                        }
```

Comment:
Let's remove all this logic and just add/update the header

```suggestion
                _.forEach(headers, function (header) {
                    // update headers which gets overwritten by the requester
                    if (OVERWRITTEN_HEADERS[_.lowerCase(header && header.key)]) {
```